x-pack/plugins/security/server/saved_objects/saved_objects_privileges.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { FeaturesService } from '../plugin';
+import {
+  isSavedObjectConditionalPrivilege,
+  SavedObjectPrivilegeCondition,
+} from '../../../features/server/feature_kibana_privileges';
+
+export class SavedObjectsPrivileges {
+  private readonly conditionalTypes: string[];
+  private readonly typesConditions: Map<
+    string,
+    Array<SavedObjectPrivilegeCondition | SavedObjectPrivilegeCondition[]>
+  >;
+
+  constructor(features: FeaturesService) {
```

Comment:
The logic within here is pretty sloppy, there are duplicates in both the `conditionalTypes` and `typesConditions`...